deepinv/datasets/datagenerator.py

```diff
@@ -129,28 +265,81 @@ def __getitem__(self, index):
                 )
                 for (k, param) in self.params.items()
             }
+        else:
+            params = None
+
+        if params is not None:
             return x, y, params
         else:
             return x, y
 
-    def __len__(self):
+    def __len__(self) -> int:
         r"""
         Returns the size of the dataset.
 
         """
-        if self.stacked > 0:
-            return len(self.y[0])
-        else:
-            return len(self.y)
+        import h5py
 
-    def close(self):
+        y = self.y
+        if not isinstance(y, h5py.Dataset):
+            y = y[0]
+        return len(y)
+
+    def close(self) -> None:
         """
         Closes the HDF5 dataset. Use when you are finished with the dataset.
         """
         if hasattr(self, "hd5") and self.hd5:
             self.hd5.close()
             self.hd5 = None
 
+    @property
+    def unsupervised(self) -> bool:
```

Comment:
wait so this `unsupervised` attribute is no longer used at all in the code? is it just for users in case they want to check the dataset? if no use surely we should deprecate it too